pycgmes/utils/base.py

```diff
@@ -129,14 +47,21 @@ def to_dict(self) -> dict[str, "CgmesAttributeTypes"]:
 
         """
         attrs = {f.name: getattr(self, f.name) for f in fields(self)}
-        attrs["__class__"] = self.resource_name
+        attrs["__class__"] = self.apparent_name()
         return attrs
 
     @cached_property
     def resource_name(self) -> str:
         """Returns the resource type."""
         return self.__class__.__name__
 
+    @cached_property
+    def namespace(self) -> str:
+        """Returns the namespace. By default, the namesapce is 'cim' for all resources.
+        Custom resources can override this.
+        """
+        return "cim"
```

Comment:
this is the prefix, though.

the actual name space is:

```suggestion
        return "http://iec.ch/TC57/CIM100#"
```

the prefix is a convention to shorten the full name. this basically means that one day i can define cim=http://iec.ch/TC57/CIM100# and the next day cim100=http://iec.ch/TC57/CIM100# and cim100:ACLineSegment and cim:ACLineSegment are the same thing because they both resolve to: http://iec.ch/TC57/CIM100#ACLineSegment

Reply:
True. This is because it is meant to be used with https://github.com/alliander-opensource/pycgmes/blob/887bfba4cd5f9a612b08ffb8105234032c510aa9/pycgmes/utils/constants.py which can be added during export. 
This is how I use it internally, but it does not feel clean indeed. 

I'll think a bit more about this, but it looks like the clean solution is indeed to give the full string as you suggest, and the prefix to long name should be managed during the actual export. After all, it's only a presentation thing.